test_tf2/test/test_message_filter.cpp

```diff
@@ -225,121 +281,138 @@ TEST(MessageFilter, multipleTargetFrames)
-{
-  BufferCore bc;
-  Notification n(1);
-  MessageFilter<geometry_msgs::PointStamped> filter(bc, "frame1", 1, 0);
-  filter.registerFailureCallback(boost::bind(&Notification::failure, &n, _1, _2));
-
-  builtin_interfaces::msg::Time stamp(1);
-  bc.setTransform(createTransform(Quaternion(0,0,0,1), Vector3(1,2,3), stamp, "frame1", "frame2"), "me");
-  bc.setTransform(createTransform(Quaternion(0,0,0,1), Vector3(1,2,3), stamp + tf2::durationFromSec(10000), "frame1", "frame2"), "me");
-
-  geometry_msgs::PointStampedPtr msg(new geometry_msgs::PointStamped);
-  msg->header.stamp = stamp;
-  msg->header.frame_id = "frame2";
-  filter.add(msg);
-
-  EXPECT_EQ(1, n.failure_count_);
-}
+  double time_stamp = (stamp.sec + stamp.nanosec/1e9) + tf2::durationToSec(offset)*1.1;
+  builtin_interfaces::msg::Time stamp_transform = rclcpp::Time((int)time_stamp, (time_stamp - (int)time_stamp)*1e9);
 
-TEST(MessageFilter, outTheBackFailure2)
-{
-  BufferCore bc;
-  Notification n(1);
-  MessageFilter<geometry_msgs::PointStamped> filter(bc, "frame1", 1, 0);
-  filter.registerFailureCallback(boost::bind(&Notification::failure, &n, _1, _2));
-
-  builtin_interfaces::msg::Time stamp(1);
+  buffer.setTransform(createTransform(tf2::Quaternion(0,0,0,1), tf2::Vector3(1,2,3), stamp_transform, "frame1", "frame2"), "me");
 
-  geometry_msgs::PointStampedPtr msg(new geometry_msgs::PointStamped);
-  msg->header.stamp = stamp;
-  msg->header.frame_id = "frame2";
-  filter.add(msg);
-
-  EXPECT_EQ(0, n.count_);
-  EXPECT_EQ(0, n.failure_count_);
-
-  bc.setTransform(createTransform(Quaternion(0,0,0,1), Vector3(1,2,3), stamp + tf2::durationFromSec(10000), "frame1", "frame2"), "me");
-
-  EXPECT_EQ(1, n.failure_count_);
-}
-
-TEST(MessageFilter, emptyFrameIDFailure)
-{
-  BufferCore bc;
-  Notification n(1);
-  MessageFilter<geometry_msgs::PointStamped> filter(bc, "frame1", 1, 0);
-  filter.registerFailureCallback(boost::bind(&Notification::failure, &n, _1, _2));
-
-  geometry_msgs::PointStampedPtr msg(new geometry_msgs::PointStamped);
-  msg->header.frame_id = "";
-  filter.add(msg);
-
-  EXPECT_EQ(1, n.failure_count_);
-}
+  EXPECT_EQ(1, n.count_); // Now have data for the message published earlier
 
-TEST(MessageFilter, callbackQueue)
-{
-  BufferCore bc;
-  Notification n(1);
-  ros::CallbackQueue queue;
-  MessageFilter<geometry_msgs::PointStamped> filter(bc, "frame1", 1, &queue);
-  filter.registerCallback(boost::bind(&Notification::notify, &n, _1));
+  time_stamp = (stamp.sec + stamp.nanosec/1e9) + tf2::durationToSec(offset);
 
-  geometry_msgs::PointStampedPtr msg(new geometry_msgs::PointStamped);
-  msg->header.stamp = builtin_interfaces::msg::Time(1);
-  msg->header.frame_id = "frame1";
+  msg->header.stamp = rclcpp::Time(time_stamp);
   filter.add(msg);
 
-  EXPECT_EQ(0, n.count_);
-
-  queue.callAvailable();
+  EXPECT_EQ(1, n.count_); // Latest message is off the end of the offset
 
-  EXPECT_EQ(1, n.count_);
+  node.reset();
 }
 
-
+// TODO(ahcorde): failurecallback
```

Comment:
Same here; can you be more descriptive on what we need to do here?

Reply:
For some unknown reason message_filters::Connection registerFailureCallback is disable with
```cpp
#if 0
  message_filters::Connection registerFailureCallback(const FailureCallback & callback)
  {
    message_connection_failure.disconnect();
    message_connection_failure = this->registerCallback(callback, this);
  }
#endif

```
https://github.com/ros2/geometry2/blob/ros2/tf2_ros/include/tf2_ros/message_filter.h#L463.

I need to rework this part when this is available